### ISSUE TITLE
Fix request: NonComparableMapKeys false positive when IntelliJ projects not fully compiled

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#469]
Better ignore imported values if possible? In this case the imported constant is of type String which is Comparable.
Also notice that the `stringToString` map is already of key type String, so should be ok!

```java
import java.util.HashMap;
import java.util.Map;
import static my.app.Constants.STRING_VALUE;

public final class IssueXXX {
    void foo() {
        Map<String, String> stringToString = new HashMap<>();
        stringToString.put(STRING_VALUE, "foo"); // <-- false positive when (static) import is used
    }
}
```

**Comment by stokpop** [comment]
should be working when `my.app.Constants.STRING_VALUE` is on classpath with `typeIs`?

**Comment by stokpop** [comment]
This false positive appeared in Intellij PMD plugin. After "rebuild" it showed that the "Lombok annotation" support was disabled. After enabling this, the false positive went away... so in IntelliJ, when code is not completely compiled or pre-generated, false positives may appear.

Can we check for that in the plugin and provide warning?

**Comment by jborgers** [comment]
Maybe detect with typeIs()? XPath element AmbigiousName?

**Comment by jborgers** [comment]
Or we can as instruction/hint in pmd plugin in intellij

**Comment by jborgers** [comment]
See #542

**Comment by jborgers** [comment]
See #568
